### DISCUSSION TITLE
AD group membership query issues

### DISCUSSION THREAD

**Discussion by pentiumoverdrive** [opendcim/openDCIM#1439] [Q&A] [answered]
Using latest OpenDCIM, I can successfully auth to AD, but it fails to pulls groups of the users.
"PHP message: LDAP authentication successful, but access denied based on lacking group membership."

Settings are:

```
LDAPBaseDN: DC=corp,DC=domain,DC=tld
LDAPBindDN: %userid%
LDAPUserSearch: (|(uid=%userid%))
LDAPBaseSearch: (|(userprincipalname=%userid%))
```

This shows all the groups
`ldapsearch  -x -h corp.domain.tld -b "DC=corp,DC=domain,DC=tld" -D "opendcimtest@corp.domain.tld" -W '(|(userprincipalname=opendcimtest@corp.domain.tld))'`

Membership:
```
memberOf: CN=OpenDCIM-Access,DC=corp,DC=domain,DC=tld
memberOf: CN=OpenDCIM-ReadOnly,DC=corp,DC=domain,DC=tld

```
Ldap access is configured as:

```
LDAPSiteAccess: CN=OpenDCIM-Access,DC=corp,DC=domain,DC=tld
LDAPReadAccess: CN=OpenDCIM-ReadOnly,DC=corp,DC=domain,DC=tld
```

Did I overlook something?

Thanks

**Comment by pentiumoverdrive** [comment]
As far as I can tell, login pulls the samaccountname or UPN of the binding user. But group search must be done with a CN= specified which is not pulled from ldap (?).

In the wiki example:

`(&(objectClass=group)(member=CN=%userid%,OU=User Location,DC=yourdomain,DC=org)) `

%userid% for my account would be "jodo" but CN would be "John Doe" so %userid% wouldn't work.

**Reply by wilpig** [reply]
I have ldap set up on the dev instance and this filter should work for the user search.  

(|(uid=%userid%)(sAMAccountName=%userid%))

Since you can't be bothered to post a full record so we can see what attributes you are using you're getting the best guess.

**Comment by pentiumoverdrive** [comment, marked as answer]
Finally got it working.

Had to use empty **LDAPBaseSearch** so login_ldap.php used this query instead:

`$query="(&(sAMAccountName=$ldapUser)(memberOf:1.2.840.113556.1.4.1941:=$groupDN))";`
